ui/tools/problemreporter/problemreporterwidget.cpp

```diff
@@ -0,0 +1,224 @@
+/*
+  problemreporterwidget.cpp
+
+  This file is part of GammaRay, the Qt application inspection and
+  manipulation tool.
+
+  Copyright (C) 2018 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Anton Kreuzkamp <anton.kreuzkamp@kdab.com>
+
+  Licensees holding valid commercial KDAB GammaRay licenses may use this file in
+  accordance with GammaRay Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "problemreporterwidget.h"
+#include "ui_problemreporterwidget.h"
+#include "problemclientmodel.h"
+#include "problemreporterclient.h"
+
+#include <ui/searchlinecontroller.h>
+#include <ui/contextmenuextension.h>
+
+#include <common/objectbroker.h>
+#include <common/objectmodel.h>
+#include <common/tools/problemreporter/problemmodelroles.h>
+
+#include <QMenu>
+#include <QStandardItemModel>
+#include <QStyledItemDelegate>
+#include <QPainter>
+#include <QMouseEvent>
+#include <QStyleOptionViewItemV4>
+
+
+#if QT_VERSION < QT_VERSION_CHECK(5, 5, 0)
+Q_DECLARE_METATYPE(Qt::CheckState)
+#endif
+
+using namespace GammaRay;
+
+class ProblemFilterDelegate : public QStyledItemDelegate
```

Comment:
this class feels like the worst code I've written this year. So, please check it extra carefully and ideally tell me, how to get the rendering right. :)